### PR TITLE
8319147: Add regression test for JDK-8317836

### PR DESCRIPTION
**Pull request by prsadhuk**
A manual regression test for JFXPanel orientation fix in [JDK-8317836](https://bugs.openjdk.org/browse/JDK-8317836) is added

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8319147](https://bugs.openjdk.org/browse/JDK-8319147): Add regression test for JDK-8317836 (**Bug** - P4)

### Reviewers
 * [Andy Goryachev](https://openjdk.org/census#angorya) (@andy-goryachev-oracle - **Reviewer**)
 * [Kevin Rushforth](https://openjdk.org/census#kcr) (@kevinrushforth - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jfx.git pull/1276/head:pull/1276` \
`$ git checkout pull/1276`

Update a local copy of the PR: \
`$ git checkout pull/1276` \
`$ git pull https://git.openjdk.org/jfx.git pull/1276/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1276`

View PR using the GUI difftool: \
`$ git pr show -t 1276`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jfx/pull/1276.diff">https://git.openjdk.org/jfx/pull/1276.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jfx/pull/1276#issuecomment-1790249053)

### CODE REVIEW

**Review by andy-goryachev-oracle — APPROVED** [review]
looks good, works as expected with the latest master.

File: tests/manual/swing/JFXPanelOrientationTest.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+import java.awt.BorderLayout;
```
Comment (kevinrushforth) [inline tests/manual/swing/JFXPanelOrientationTest.java:25]: Minor: we prefer a blank line between the copyright block and the package or import statements.

File: tests/manual/swing/JFXPanelOrientationTest.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+import javafx.scene.Scene;
+import javafx.stage.Stage;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.TextArea;
+import javafx.scene.layout.BorderPane;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.VBox;
+
+
```
Comment (kevinrushforth) [inline tests/manual/swing/JFXPanelOrientationTest.java:46]: Minor: you can remove the extra blank line.

**Review by kevinrushforth — APPROVED** [review]
I verified that this test passes with the fix for JDK-8317836 and fails with that fix locally reverted.

The only minor thing I noticed is that the instruction window is obscured by the test window, so you have to move the test window out of the way to see it. Maybe set the X and Y of the instruction window and/or make the test window smaller so they don't overlap?

If you would like to fix this, I also left two very minor code format comments that you might consider. I'll approve anyway, and reapprove if you make the changes.

**Comment by prsadhuk** [comment]
> make the test window smaller so they don't overlap
Done

> Maybe set the X and Y of the instruction window
Although it is probably not required now, but can you please specify how to do it as I dont see Scene accepting x,y params. Is there any in-house or public fx documentation that I can peruse (guess we dont have similar to https://docs.oracle.com/en/java/javase/21/docs/api/index.html)

**Comment by andy-goryachev-oracle** [comment]
> can you please specify how to do it for my future reference as I dont see Scene accepting x,y params

Use Stage (Window)  setX(), setY() before show().

**Comment by kevinrushforth** [comment]
> Is there any in-house or public fx documentation that I can peruse (guess we dont have similar to https://docs.oracle.com/en/java/javase/21/docs/api/index.html)

Andy already answered your specific question. As for where the docs are, you can find them either by going to [jdk.java.net/javafx21](https://jdk.java.net/javafx21/) and click on the API docs link, which points here:

https://download.java.net/java/GA/javafx21.0.1/e5ab43c6aed54893b0840c1f2dcfca4d/docs/api/overview-summary.html

Or go to [openjfx.io](https://openjfx.io/) and clicking on the API docs link, which points here:

https://openjfx.io/javadoc/21/

**Review by kevinrushforth — APPROVED** [review]

**Comment by kevinrushforth** [comment]
> I was waiting for changes to set the window position, but can re-approve if necessary.

His earlier comments suggested that he felt it wasn't needed for this PR; with the adjusted window sizes, the windows no longer overlap.

> A side comment: the arrangement of the test instruction window (as well as the creating the said window) should be a convenience method somewhere in common test units, should it not?

Good idea. Can you file an enhancement to consider this for the next test sprint?

**Comment by andy-goryachev-oracle** [comment]
> Can you file an enhancement to consider this for the next test sprint?

Created JDK-8319555

**Comment by prsadhuk** [comment]
Thank you for the FX API docs link

**Pull request closed without merging**